src/server/app_render.jsx

```diff
@@ -16,7 +16,10 @@ async function appRender(ctx) {
     const store = {};
     try {
         let login_challenge = ctx.session.login_challenge;
-        let user_preferences = null;
+        let locale = ctx.getLocaleFromHeader();
+        if (locale) locale = locale.substring(0, 2);
+        if (!locale.match(/(en|es|ru)/)) locale = 'en';
```

Comment:
i think the list of supported locales should be autogenerated on server startup from the locale files available in the directory and not hardcoded into a regex buried in the middle of app_render.jsx. at the very least they should be a list in `$STM_Config`.